src/com/cdd/bao/axioms/CellLineFix.java

```diff
@@ -0,0 +1,531 @@
+		   a contrived parent rooted at the BRENDA root.  For example, we might have contrived
+		   parents that house cell- or tissue-related terms only.
+
+	The result of this merging operation -- a set of semantic directives -- is written to a
+	specified file meant to be curated at a later time by a human.
+*/
+
+public class CellLineFix 
+{
+	private Vocabulary vocab = null;
+	
+	private static class CellPair
+	{
+		String uri1, label1;
+		String uri2, label2;
+	}
+
+	private static class Match
+	{
+		String uri, label;
+		int score;
+	}
+
+	private final static String URI_BRENDA_ROOT = "http://dto.org/DTO/DTO_000";
+	private final static String URI_CLO_ROOT = "obo:CLO_0000001";
+	
+	// XXX contrived branches rooted at CLO and used to house BRENDA cell and tissue terms
+	// XXX note these closely resemble provisional terms in BAE
+	private final static String URI_FOR_CELL_LINES = "obo:CLO_9999998";
+	private final static String URI_FOR_TISSUES = "obo:CLO_9999999";
+
+	private List<CellPair> pairs = new ArrayList<>();
+	private Set<String> skipSet = new HashSet<>();
+	private Map<String, String> brendaMap = new HashMap<>(), cloMap = new HashMap<>(); // uri-to-label
+	private Set<String> brendaMatched = new HashSet<>(); // set of BRENDA terms that matched at least one CLO term
+
+	private PrintWriter outWriter;
+	private String curationFN;
+	private int maxScore;
+
+	public CellLineFix(File outfile, String curationFN, int maxScore) throws FileNotFoundException
+	{
+		this.outWriter = new PrintWriter(outfile);
+		this.curationFN = curationFN;
+		this.maxScore = maxScore;
+	}
+
+	public void exec() throws Exception
+	{
+		loadCellPairs();
+		Util.writeln("Predefined pairs: " + pairs.size() + ", skip: " + skipSet.size());
+
+		Util.writeln("Loading vocab...");
+		vocab = new Vocabulary("./data/ontology", null);
+		Util.writeln("... properties: " + vocab.numProperties() + ", values: " + vocab.numValues());		
+		
+		Vocabulary.Hierarchy hier = vocab.getValueHierarchy();
+		Vocabulary.Branch brendaRoot = hier.uriToBranch.get(ModelSchema.expandPrefix(URI_BRENDA_ROOT));
+		Vocabulary.Branch cloRoot = hier.uriToBranch.get(ModelSchema.expandPrefix(URI_CLO_ROOT));
+		
+		for (Vocabulary.Branch br : brendaRoot.children) flattenBranch(brendaMap, br, true);
+		for (Vocabulary.Branch br : cloRoot.children) flattenBranch(cloMap, br, false);
+		
+		Util.writeln("BRENDA terms: " + brendaMap.size());
+		Util.writeln("CLO terms:    " + cloMap.size());
+		
+		for (CellPair cell : pairs)
+		{
+			brendaMap.remove(cell.uri1);
+			brendaMap.remove(cell.uri2);
+			cloMap.remove(cell.uri1);
+			cloMap.remove(cell.uri2);
+		}
+		
+		// iterate over remaining CLO pairs, look for possible BRENDA matches
+		int count = 0;
+		for (Map.Entry<String, String> entry : cloMap.entrySet())
+		{
+			String cloURI = entry.getKey(), cloLabel = entry.getValue();
+			if (skipSet.contains(cloURI)) continue;
+			if (++count > 64) {Util.writeln(""); count = 0;}
+			Util.write(".");
```

Comment:
I usually use `writeFlush` for this kind of thing, output is often buffered.

Reply:
The `....` is just a poor man's progress meter on a line terminal :-)

Reply:
Indeed, but normal console output won't write any of the dots until it gets to newline or buffer size... so if you run it outside of Eclipse, the progress won't be very helpful, hence `writeFlush`.